src/js/common/schemaform/widgets/CurrencyWidget.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+export default class CurrencyWidget extends React.Component {
+  onBlur = () => {
+    this.props.onBlur(this.props.id);
+  }
+
+  handleChange = (event) => {
+    const val = event.target.value;
+    if (val === '') {
+      this.props.onChange();
+    } else {
+      // Sometimes we get a string back (e.g. when you enter 10.00)
+      // so we need to parse it to be sure
+      const parsed = parseFloat(val);
+      if (!isNaN(parsed)) {
+        this.props.onChange(parsed);
+      } else {
+        this.props.onChange(val);
```

Comment:
Oh, I guess it would be a browser-specific functionality, but I was thinking since they _are_ number inputs, we can't even enter non-numbers. Good point, though.